ignite/base/events_driven.py

```diff
@@ -257,34 +262,35 @@ class EventsDrivenState:
     Args:
         engine: ignite engine :class:`~ignite.base.events_driven.EventsDriven` that used to access
             the allowed events and their counters.
-        event_to_attr: mapping consists of the events from :class:`~ignite.engine.events.Events`
-            or any other custom events added by :meth:`~ignite.base.events_driven.EventsDriven.register_events`.
+        attr_to_events: mapping consists of the attributes mapped to a list events from
+            :class:`~ignite.engine.events.Events` or any other custom events added
+            by :meth:`~ignite.base.events_driven.EventsDriven.register_events`.
         **kwargs: optional keyword args.
 
+    Note:
+        The attributes will be retrieved according to the first element in the list of the event at `attr_to_events`.
     """
 
     def __init__(
-        self, engine: Optional[EventsDriven] = None, event_to_attr: Optional[Mapping[Any, str]] = None, **kwargs: Any
+        self,
+        engine: Optional[EventsDriven] = None,
+        attr_to_events: Optional[Mapping[str, List["Events"]]] = None,
+        **kwargs: Any,
     ):
-        if event_to_attr is not None and engine is None:
-            raise ValueError("Both engine and event_to_attr should be provided, but only event_to_attr is given")
 
         self.engine = engine  # type: Optional[EventsDriven]
-
-        self._attr_to_events = defaultdict(list)  # type: Mapping[str, Any]
-        if event_to_attr is not None:
-            for k, v in event_to_attr.items():
-                self._attr_to_events[v].append(k)
+        self._attr_to_events = defaultdict(list)  # type: Mapping[str, List[Events]]
+        if attr_to_events:
+            self._attr_to_events = attr_to_events
```

Comment:
Recode this lines into a single line definition

Reply:
Check also for the given type and raise issues if something else is given

Reply:
@KickItLikeShika why did you resolve the comments without updating the code ?

Reply:
Oh sorry my bad, I will update it now